src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/service/BookedLocationsService.kt

```diff
@@ -16,40 +25,87 @@ class BookedLocationsService(
-          )
+        .mapNotNull { result ->
+          // Only include if result does not match existing BVLS appointment
+          if (existingAppointments.none { existing -> result.matches(existing) }) {
+            BookedLocation(
+              locations.single { l -> l.dpsLocationId == nomisToDpsLocations[result.internalLocation!!.id] },
+              startTime = LocalTime.parse(result.startTime),
+              endTime = LocalTime.parse(result.endTime),
+            )
+          } else {
+            null
+          }
         }
         .let { BookedLocations(it) }
 
       false -> prisonApiClient.getScheduledAppointments(prisonCode, date, nomisToDpsLocations.keys)
-        .map { result ->
-          BookedLocation(
-            locations.single { l -> l.dpsLocationId == nomisToDpsLocations[result.locationId] },
-            startTime = result.startTime.toLocalTime(),
-            endTime = result.endTime!!.toLocalTime(),
-          )
+        .mapNotNull { result ->
+          // Only include if result does not match existing BVLS appointment
+          if (existingAppointments.none { existing -> result.matches(existing) }) {
+            BookedLocation(
+              locations.single { l -> l.dpsLocationId == nomisToDpsLocations[result.locationId] },
+              startTime = result.startTime.toLocalTime(),
+              endTime = result.endTime!!.toLocalTime(),
+            )
+          } else {
+            null
+          }
         }
         .let { BookedLocations(it) }
     }
   }
+
+  private fun AppointmentSearchResult.matches(prisonAppointment: PrisonAppointment) = isTheSameTime(prisonAppointment) &&
+    isTheSameAppointmentType(supportedAppointmentTypes.typeOf(prisonAppointment.bookingType())) &&
+    attendees.any { it.prisonerNumber == prisonAppointment.prisonerNumber }
+
+  private fun ScheduledAppointment.matches(prisonAppointment: PrisonAppointment) = isTheSameTime(prisonAppointment) &&
+    isTheSameAppointmentType(supportedAppointmentTypes.typeOf(prisonAppointment.bookingType())) &&
+    offenderNo == prisonAppointment.prisonerNumber
 }
 
+data class BookedLookup(
+  val prisonCode: String,
+  val date: LocalDate,
+  val locations: Collection<Location>,
+  val videoBookingIdToExclude: Long? = null,
+)
+
 data class BookedLocation(val location: Location, val startTime: LocalTime, val endTime: LocalTime)
 
 class BookedLocations(booked: Collection<BookedLocation>) {
 
-  private val locations = booked.groupBy { it.location.key }
+  private val locations = booked.groupBy { it.location.dpsLocationId }
 
-  fun isBooked(location: Location, startTime: LocalTime, endTime: LocalTime) = locations[location.key]?.any { isTimesOverlap(it.startTime, it.endTime, startTime, endTime) } ?: false
+  fun isBooked(location: Location, startTime: LocalTime, endTime: LocalTime) = locations[location.dpsLocationId]?.any { isTimesOverlap(it.startTime, it.endTime, startTime, endTime) } ?: false
```

Comment:
Changed to use the ID instead of key for the location, although using the key still works.

Should be avoiding use of the key really where possible.